config/initializers/zeitwerk.rb

```diff
@@ -11,9 +11,10 @@
   Rails.autoloaders.main.collapse(Rails.root.join("lib/ansible/runner/credential"))
   Rails.autoloaders.main.collapse(Rails.root.join("lib/pdf_generator"))
 
-  # requires qpid, which is an optional dependency because LOL mac
-  if RUBY_PLATFORM.include?("darwin")
-    message_handler_path = Pathname.new(Vmdb::Plugins.paths[ManageIQ::Providers::Nuage::Engine]).join("app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb")
-    Rails.autoloaders.main.ignore(message_handler_path)
+  # These stub classes are only for macos to workaround qpid_proton not being installable on mac.
+  # These stubs won't define constants elsewhere, so we need to tell zeitwerk to ignore them.
+  unless RUBY_PLATFORM.include?("darwin")
+    Rails.autoloaders.main.ignore(Rails.root.join("lib/qpid/proton/container.rb"))
+    Rails.autoloaders.main.ignore(Rails.root.join("lib/qpid/proton/messaging_handler.rb"))
```

Comment:
I don't like it but if we're conditionally defining classes in lib/qpid/proton/* below, we need to tell the autoloaders to ignore them on platforms that will not define them as zeiwerk will complain about them.

This should fix `DEBUG_MANAGEIQ_ZEITWERK=true bin/rails zeitwerk:check --trace` on linux @agrare

Reply:
```
NameError: uninitialized constant Qpid
/home/grare/adam/.gem/ruby/3.0.0/bundler/gems/manageiq-providers-nuage-541494cded64/app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb:1:in `<main>'
```

Reply:
Yeah I don't know, I have `BUNDLE_WITH=qpid_proton` and it is in my Gemfile.lock
```
$ grep BUNDLE_WITH .bundle/config 
BUNDLE_WITH: "qpid_proton,systemd"
$ grep qpid_proton Gemfile.lock 
    qpid_proton (0.30.0)
  qpid_proton (~> 0.30.0)
$ DEBUG_MANAGEIQ_ZEITWERK=true bin/rails zeitwerk:check --trace
NameError: uninitialized constant Qpid
/home/grare/adam/.gem/ruby/3.0.0/bundler/gems/manageiq-providers-nuage-541494cded64/app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb:1:in `<main>'
<internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:38:in `require'
<internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:38:in `require'
/home/grare/adam/.gem/ruby/3.0.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require
```
and I merged this PR locally
```
commit 5803d9e827854b4662cec6569185b04d610faff7 (HEAD -> master)
Merge: 883525adba 3b3f192068
Author: Adam Grare <adam@grare.com>
Date:   Wed Oct 25 11:10:27 2023 -0400

    Merge pull request #22752 from jrafanie/fix_qpid_proton_require_under_zeitwerk
    
    Fix qpid proton require under zeitwerk

commit 3b3f192068fd41c3f235912f4a4bfd3bfdcdaab0
Author: Joe Rafaniello <jrafanie@gmail.com>
Date:   Mon Oct 23 16:11:15 2023 -0400
```

Reply:
Making this up:

The error is due to zeitwerk wanting to define the constant qpid, finding the file `qpid.rb` and requiring it.
I bet they expect it to work just fine

Speculation:

If @agrare were to delete those 2 files, it would work just fine

Maybe we could zeitwerk ignore those files on all but mac?

Reply:
sorry ^ is useless and obvious. ignore

Reply:
ok. so these files (aka defining these constants) are only required for development on a mac.
Wonder if we want to add an initializer that defines these 2 constants on the mac?

Reply:
@agrare can you try this:


```
BUNDLER_GROUPS="manageiq_default,ui_dependencies,qpid_proton" DEBUG_MANAGEIQ_ZEITWERK=true bin/rails zeitwerk:check --trace
```

We weren't auto-requiring qpid_proton during rails boot because qpid_proton wasn't in the default bundler groups.

Reply:
With `rails c` in core:
```
>> require 'qpid_proton'
=> true
```

```
$ BUNDLER_GROUPS="manageiq_default,ui_dependencies,qpid_proton" DEBUG_MANAGEIQ_ZEITWERK=true bin/rails zeitwerk:check --trace
...
NameError: uninitialized constant Qpid
/home/grare/adam/.gem/ruby/3.0.0/bundler/gems/manageiq-providers-nuage-541494cded64/app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb:1:in `<main>'
<internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:38:in `require'
<internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:38:in `require'
/home/grare/adam/.gem/ruby/3.0.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
```